src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -96,6 +96,12 @@ trait GenTraversableOnce[+A] extends Any {
    */
   def size: Int
 
+   /** The size of this $coll if it is can be cheaply computed
+   *
+   *  @return    the number of elements in this $coll, or -1 if the size cannot be determined cheaply
+   */
+  protected[collection] def sizeHint: Int = -1
```

Comment:
Btw, using `sizeHint` for the name of this method looks like a very idea to me:
* It does not have the same meaning as the other methods called `sizeHint` (which are *also* in the hierarchy): this one is a "getter" of sorts, while the others are "setters".
* It creates an overload between a `()`-less `def` and other defs *with* `()`. I've always found this was not very clean (and, IIRC, it's not even allowed in dotty anymore)

Can we find another name?